Make (version 4.4-1) command is not creating a temp directory

# Problem

If I checkout a new branch and run `make run-clojure` then this error is thrown:

```
[status-mobile]-> make run-clojure                                                            
yarn shadow-cljs watch mobile                                                                 
Configuring Nix shell for target 'clojure'...                                                 
error: getting status of '/tmp/tmp-status-mobile-1cdcd298b': No such file or directory        
make: *** [Makefile:260: run-clojure] Error 1
```

Apparently `_tmpdir-mk` target is not being called. We have to call it manually `make _tmpdir-mk`

(Creating issue for context, if anyone else gets this error)

> What if you add -d flag and grep for _tmpdir-mk, like so:

```
[status-mobile]-> make run-clojure -d | grep _tmpdir-mk
Reading makefile '_tmpdir-mk' (search path) (don't care) (no ~ expansion)...
Makefile '_tmpdir-mk' might loop; not remaking it.
Configuring Nix shell for target 'clojure'...
error: getting status of '/tmp/tmp-status-mobile-1cdcd298b': No such file or directory
make: *** [Makefile:260: run-clojure] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-.PHONY: nix-add-gcroots clean nix-clean run-metro test release _list _fix-node-perms _tmpdir-mk _tmpdir-rm _install-hooks
+.PHONY: nix-add-gcroots clean nix-clean run-metro test release _list _fix-node-perms _tmpdir-rm
 
 help: SHELL := /bin/sh
 help: ##@other Show this help
@@ -118,10 +118,11 @@ _fix-node-perms: ##@prepare Fix permissions so that directory can be cleaned
 	$(shell test -d node_modules && chmod -R 744 node_modules)
 	$(shell test -d node_modules.tmp && chmod -R 744 node_modules.tmp)
 
-_tmpdir-mk: SHELL := /bin/sh
-_tmpdir-mk: ##@prepare Create a TMPDIR for temporary files
+$(TMPDIR): SHELL := /bin/sh
+$(TMPDIR): ##@prepare Create a TMPDIR for temporary files
 	@mkdir -p "$(TMPDIR)"
 # Make sure TMPDIR exists every time make is called
+_tmpdir-mk: $(TMPDIR)
 -include _tmpdir-mk
 
 _tmpdir-rm: SHELL := /bin/sh

